src/Symfony/Component/JsonPath/Functions/ValueFunction.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonPath\Functions;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+#[AsJsonPathFunction(name: 'value')]
+final class ValueFunction extends AbstractJsonPathFunction implements JsonPathComparableFunctionInterface
+{
+    use JsonPathFunctionArgumentTrait;
+
+    public function __invoke(array $args, mixed $context): mixed
+    {
+        $results = $args[0] ?? [];
+        if (!\is_array($results)) {
+            return $results;
+        }
+
+        $nodelistSize = \count($results);
+        if (1 < $nodelistSize) {
```

Comment:
we could remove this check, using only `return 1 === $nodelistSize ? $results[0] : null;` for the same result.